### PR TITLE
This branch is fixups for the cni-plugin branch

### PR DESCRIPTION
**Pull request by stevej**
This branch is off of `stevej/cni-plugin-target` and will contain all the messy work to get that branch to green.

This branch MUST be squashed into that branch so that branch can be merged into `main`.

### CODE REVIEW

**Comment by olix0r** [comment]
I had no problems moving the shared code into `internal/` https://github.com/linkerd/linkerd2-proxy-init/pull/148

File: cni-plugin/main.go [hunk 1]
```diff
@@ -230,7 +241,7 @@ func cmdAdd(args *skel.CmdArgs) error {
 			}
 
 			// Override ProxyUID from annotations.
-			proxyUIDOverride, err := getAnnotationOverride(ctx, client, pod, k8s.ProxyUIDAnnotation)
+			proxyUIDOverride, err := getAnnotationOverride(ctx, client, pod /* k8sProxyUIDAnnotation */, "true")
```
Comment (alpeb) [inline cni-plugin/main.go:244]:
```suggestion
			proxyUIDOverride, err := getAnnotationOverride(ctx, client, pod)
```

File: cni-plugin/main.go [hunk 2]
```diff
@@ -298,7 +310,7 @@ func cmdDel(args *skel.CmdArgs) error {
 	return nil
 }
 
-func getAnnotationOverride(ctx context.Context, api *k8s.KubernetesAPI, pod *v1.Pod, key string) (string, error) {
+func getAnnotationOverride(ctx context.Context, api *kubernetes.Clientset, pod *v1.Pod, key string) (string, error) {
 	// Check if the annotation is present on the pod
 	if override := pod.GetObjectMeta().GetAnnotations()[key]; override != "" {
 		return override, nil
```
Comment (alpeb) [inline cni-plugin/main.go:316]:
```suggestion
func getAnnotationOverride(ctx context.Context, api *kubernetes.Clientset, pod *v1.Pod) (string, error) {
	// Check if the annotation is present on the pod
	if override := pod.GetObjectMeta().GetAnnotations()["config.linkerd.io/skip-outbound-ports"]; override != "" {
		return override, nil
```

**Review by alpeb — COMMENTED** [review]
Looking good :+1: 

Note that with the `iptables_test.go` and `portrange_test.go` moves, the `proxy-init-test-unit` recipe is no longer triggering those tests.

Also, It seems there was an oversight in the `cni-plugin/main.go` refactoring, so I added a suggestion below.

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Pull request merged**